### ISSUE TITLE
Failed to increase pv size after successful increase of pvc size

### ISSUE THREAD

**Issue by SidAtQuanos** [hetznercloud/csi-driver#614]
### TL;DR

Hi there,

the csi driver failed to increase the persistent volumes after redeployment of a helm chart (elastic) with increased data storage claims for the pvc

### Expected behavior

Hetzner Volume size is increased: success
PVC Volume size increased: success
PV Volume size increased: success
Disk resize: success

### Observed behavior

Hetzner Volume size is increased: success
PVC Volume size increased: success
PV Volume size increased: **failed**
Disk resize: **failed**

Error message
resize volume "pvc-<id>" by resizer "csi.hetzner.cloud" failed: rpc error: code = Internal desc = failed to expand volume: volume size is to small (invalid_input)

In this case: hetzner volume was increased to 40 GI, the persistent volume still was 30 Gi, so the csi-driver was not able to run resize2fs successfully (as 30 Gi is smaller than 40 Gi)

### Additional information

A prior update (also using helm, updating the data storage... from 20 to 30 Gi) worked for 3 nodes but failed for 3 other (out of 6). We updated the pv of the failing 3 manually. Now all 6 nodes failed.

Kubernetes Version: 1.29.1
csi-driver version: 2.6.0

Sadly it's a few days back in the past and we lag further details from the kubernetes cluster and can't look up additional details. If required we might try to update from 40Gi to possibly 45Gi to see if we can reproduce the failure

**Comment by jooola** [comment]
This might be related to https://github.com/hetznercloud/csi-driver/pull/623

Could you tell us if upgrading to v2.7.1 fixed the problem?
